libsupport/include/katana/EntityTypeManager.h

```diff
@@ -78,6 +83,181 @@ class EntityTypeManager {
         atomic_entity_type_id_to_entity_type_ids_(
             std::move(atomic_entity_type_id_to_entity_type_ids)) {}
 
+  template <typename ArrowType>
+  struct PropertyColumn {
+    int field_index;
+    std::shared_ptr<ArrowType> array;
+
+    PropertyColumn(int i, std::shared_ptr<ArrowType>& a)
+        : field_index(i), array(a) {}
+  };
+
+  /// This function can be used to convert "old style" graphs (storage format 1,
+  /// where types are represented by bool or uint8 properties) and "new style"
+  /// graphs (version > 2, where types are represented in our native type
+  /// represenation). This function is serial but it likely iterates over
+  /// O(nodes) and O(edges) vectors, so it is very slow. It should only be used
+  /// for updating old graphs.
+  ///
+  /// The number of entries in entity_type_ids should be equal to topo_size and
+  /// properties->num_rows().
+  template <template <typename> class Vector>
+  static Result<void> AssignEntityTypeIDsFromProperties(
+      const size_t topo_size,  // == either num_nodes() or  num_edges()
+      const std::shared_ptr<arrow::Table>& properties,
+      katana::EntityTypeManager* entity_type_manager,
+      Vector<EntityTypeID>* entity_type_ids) {
+    KATANA_LOG_ASSERT(entity_type_ids->size() == topo_size);
+    int64_t num_rows = properties->num_rows();
+    if (num_rows == 0) {
+      std::fill(
+          entity_type_ids->begin(), entity_type_ids->end(), kUnknownEntityType);
+      return ResultSuccess();
+    }
+    auto num_rows_for_comparison = static_cast<size_t>(num_rows);
+    KATANA_LOG_VASSERT(
+        entity_type_ids->size() == num_rows_for_comparison,
+        "size: {}, expected: {}, signed expected: {}", entity_type_ids->size(),
+        num_rows_for_comparison, num_rows);
```

Comment:
This needs to return an errors instead of being an assert since a caller can trigger this failure.

Reply:
Will do